app/cdap/components/Connections/index.tsx

```diff
@@ -121,7 +121,7 @@ export default function Connections({
         const disabledTypes = config?.connectionConfig?.disabledTypes;
         const def = config?.connectionConfig?.defaultConnection;
 
-        if (def && allowDefaultConnection && !connectionId) {
+        if (def && getCurrentNamespace() === 'default' && allowDefaultConnection && !connectionId) {
```

Comment:
Is it possible that the default connection is in a different namespace?

Reply:
That is, say the default connection exists in namespace `foo` and the user is currently in namespace `bar` (or `default`). Is that possible? If so, it seems like we have to find a place to check for the existence of the namespace/connection combo.

Reply:
Yaojie said it is not currently possible. Only the default namespace can have a default connection right now.